### PR TITLE
feat(filter) replace select by select2 for service by host

### PR DESCRIPTION
**Pull request by loiclau**
# Pull Request Template

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/configuration/configObject/service/listServiceByHost.php [hunk 1]
```diff
@@ -71,6 +71,7 @@
     $_POST['searchS'] ?? $_GET['searchS'] ?? null,
     FILTER_SANITIZE_STRING
 );
+var_dump($template);
```
Comment (loiclau) [inline www/include/configuration/configObject/service/listServiceByHost.php]:
```suggestion
```

File: www/include/configuration/configObject/service/listServiceByHost.php [hunk 2]
```diff
@@ -82,17 +83,18 @@
...
-    $template = $centreon->historySearch[$url]['template'] ?? null;
+    $template = $centreon->historySearch[$url]['template'] ?? 0;
     $searchH = $centreon->historySearch[$url]["searchH"] ?? null;
     $searchS = $centreon->historySearch[$url]["searchS"] ?? null;
     $hostStatus = $centreon->historySearch[$url]["hostStatus"] ?? 0;
-    $status = $centreon->historySearch[$url]["status"] ?? -1;
+    $status = $centreon->historySearch[$url]["status"] ?? null;
 }
 
+var_dump($template);
```
Comment (loiclau) [inline www/include/configuration/configObject/service/listServiceByHost.php]:
```suggestion
```

File: www/include/configuration/configObject/service/listServiceByHostGroup.php [hunk 3]
```diff
@@ -247,8 +225,27 @@
...
+$attrServiceStatus = null;
+if ($status) {
+    $statusDefault = array($statusFilter[$status] => $status);
+    $attrServiceStatus = array(
+        'defaultDataset' => $statusDefault
+    );
+}
+$form->addElement('select2', 'status', "", $statusFilter, $attrServiceStatus);
 
+// Fill a tab with a multidimensional Array we put in $tpl
```
Comment (kduret) [inline www/include/configuration/configObject/service/listServiceByHostGroup.php]: this comment is relevant ?

Comment (loiclau) [inline www/include/configuration/configObject/service/listServiceByHostGroup.php]:
```suggestion
```

**Pull request merged**